cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -105,27 +138,52 @@ private URI addCookieInterceptorIfRequired(URI uri) {
                 path = path.substring(0, index);
             }
 
-            URI baseURI = new URI(uriProtocol, null, uriHost, uriPort, path, null, null);
-            CookieInterceptor ci = new CookieInterceptor(this.username, this.password, baseURI.toString());
-            requestInterceptors.add(ci);
-            responseInterceptors.add(ci);
+            return new URI(uriProtocol, null, uriHost, uriPort, path, null, null);
         } catch (URISyntaxException e) {
             throw new RuntimeException(e);
         }
+    }
 
+    private URL getIamSessionUrl() {
+        // TODO allow over-riding with a system property
```

Comment:
This needs to be added, although given we have the same over-ride in java-cloudant maybe we could move it into the interceptor itself rather than needing to specify it in the constructor?

Reply:
Pushed this into the java-cloudant project and fixed up in 9d065777